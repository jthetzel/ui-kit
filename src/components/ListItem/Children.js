import styled from 'react-emotion';

export default styled.div`
  margin-top: 10px;
`;
