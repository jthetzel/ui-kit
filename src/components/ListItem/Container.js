import styled from 'react-emotion';

const hover = `
  &:hover {
    color: #223bf2;
    background-color: #faf9fa;
  }
`;

export default styled.div`
  font-family: 'Noto Sans', sans-serif;
  padding: 15px 20px;
  background-color: ${({ isActive }) => (isActive ? '#f5f4f5' : '#ffffff')};
  border-bottom: 1px solid #ecf1f8;
  cursor: pointer;
  transition: 0.5s;

  ${({ isActive }) => (!isActive && hover)};
`;
