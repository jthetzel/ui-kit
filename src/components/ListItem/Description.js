import styled from 'react-emotion';

export default styled.div`
  font-size: 12px;
  color: #303d41;
  margin: 5px 0;
`;
