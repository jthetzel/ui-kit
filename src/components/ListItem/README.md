ListItem

```js
initialState = { active: 2 };

const options = [
  { value: 'one', label: 'One' },
  { value: 'two', label: 'Two' },
  { value: 'three', label: 'Three' }
];

const style = {
  width: '300px',
  border: '1px solid #999'
};

<div style={style}>
    <ListItem
      title={'Плотность населения'}
      isActive={state.active === 0}
      onClick={() => setState({ active: 0 })}
    />
    <ListItem
      title={'Удобство для жизни'}
      description={'Доступность инфраструктуры и сервисов в пешеходной доступности жилых домов'}
      units={'km/2'}
      isActive={state.active === 1}
      onClick={() => setState({ active: 1 })}
    />
    <ListItem
      title={'Транспортная доступноть (PTAL)'}
      description={'Коэффициент доступности территории на общественном транспорте'}
      isActive={state.active === 2}
      onClick={() => setState({ active: 2 })}
    >
      <Slider
        options={options}
        onChange={(value) => value}
      />
    </ListItem>
</div>
```