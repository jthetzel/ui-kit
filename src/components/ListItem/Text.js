import styled from 'react-emotion';

export default styled.div`
  font-size: 12px;
`;
