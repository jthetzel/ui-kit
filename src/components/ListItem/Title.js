import styled from 'react-emotion';

export default styled.div`
  font-size: 16px;
  font-weight: ${({ isActive }) => (isActive ? 'bold' : 'inherit')};
  margin: 5px 0;
  color: ${({ isActive }) => (isActive ? '#303d41' : 'inherit')};
`;
