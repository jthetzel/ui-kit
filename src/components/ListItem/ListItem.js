import React from 'react';
import PropTypes from 'prop-types';

// Style
import Container from './Container';
import Title from './Title';
import Description from './Description';
import Units from './Units';

/**
 * @component
 */
const ListItem = (props) => {
  const {
    title,
    description,
    units,
    children
  } = props;

  return (
    <Container>
      <Title>{title}</Title>
      {
        description &&
          <Description>{description}</Description>
      }
      {
        units &&
          <Units>{units}</Units>
      }
      {children}
    </Container>
  );
};

ListItem.propTypes = {
  title: PropTypes.node.isRequired,
  description: PropTypes.node,
  units: PropTypes.node,
  children: PropTypes.node
};

ListItem.defaultProps = {
  description: PropTypes.null,
  units: PropTypes.null,
  children: PropTypes.null
};

export default ListItem;
