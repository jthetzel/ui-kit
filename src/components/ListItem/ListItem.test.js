import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ListItem from '../ListItem';

test('ListItem does not crash', () => {
  const tree = shallow(<ListItem
    title="test"
    description="test"
    onClick={() => {}}
  />);
  expect(toJson(tree)).toMatchSnapshot();
});

test('ListItem test isActive', () => {
  const wrapper = mount(<ListItem
    title="test"
    description="test"
    onClick={() => {}}
    isActive={false}
  />);
  expect(wrapper.props().isActive).toEqual(false);
  wrapper.setProps({ isActive: true });
  expect(wrapper.props().isActive).toEqual(true);
});
