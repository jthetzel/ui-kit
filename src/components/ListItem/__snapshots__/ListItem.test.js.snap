// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem does not crash 1`] = `
<Styled(div)
  isActive={false}
  onClick={[Function]}
>
  <Styled(div)
    isActive={false}
  >
    test
  </Styled(div)>
  <Styled(div)
    isActive={false}
  >
    test
  </Styled(div)>
</Styled(div)>
`;
